### PR TITLE
Add ERA5 Monthly Single Levels 1979-2024 artifact

### PR DESCRIPTION
**Pull request by imreddyTeja**
The issue [here](https://github.com/CliMA/ClimaAtmos.jl/issues/3424) has conflicting info on if there should be separate files for atmos and land calibration. The issue says "this can be in the same file but land wont use it" and 
"We can have separate data files for atmos and land calibration. Only surface fluxes variables will be shared, and they can stay in the land data file."

I currently have the variables in one file. I thought this made more sense because only one variable is not used by land, so the atmos calibration data would contain a single variable. 

Checklist:
- [x] I created a new folder `$artifact_name`
  - [x] I added a `README.md` in that that folder that
    - [x] describes the data and processing done to it
    - [x] lists the sources of the raw data
    - [x] lists the required citation, licenses
  - [x] If applicable (e.g., for Creative Commons), I added a `LICENSE` file
  - [x] I added the scripts that retrieve, process, and produce the artifact
  - [x] I added the environment used for such scripts (typically, `Project.toml`
        and `Manifest.toml`)
  - [x] I added the `OutputArtifacts.toml` file containing the information
        needed for package developers to add `$artifact_name` to their package
- [x] I uploaded the artifact folder to the Caltech cluster (in
      `/groups/esm/ClimaArtifacts/artifacts/$artifact_name`)
- [x] I added the relevant code to the `Overides.toml` on the Caltech Cluster
      (in `/groups/esm/ClimaArtifacts/artifacts/Overrides.toml`)
- [x] I added a link to the main `README.md` to point to the new artifact

Plots of monthly mean:
mslhf_map
msror_map
msshf_map
mssror_map
msuwlwrf_map
msuwswrf_map
tcw_map

Plots of monthly means by hour of day (these plots are from 23:00-24:00:

mslhf_hourly_map
msror_hourly_map
msshf_hourly_map
mssror_hourly_map
msuwlwrf_hourly_map
msuwswrf_hourly_map
tcw_hourly_map

### CODE REVIEW

**Comment by szy21** [comment]
I think it would be better to have total column water in a separate file. We will likely add more atmos variables to that file, or replace it with something else. But we can also do this later.

**Comment by kmdeck** [comment]
> I think it would be better to have total column water in a separate file. We will likely add more atmos variables to that file, or replace it with something else. But we can also do this later.

Was there also a question of having surface variables in one file (some used by land, some by atmos), but 3d vars in a separate one only for atmos?

**Comment by szy21** [comment]
> > I think it would be better to have total column water in a separate file. We will likely add more atmos variables to that file, or replace it with something else. But we can also do this later.
> 
> Was there also a question of having surface variables in one file (some used by land, some by atmos), but 3d vars in a separate one only for atmos?

Yeah I was thinking surface variables in one file, 2d variables (e.g, vertical integrals such as water vapor path) for atmosphere only in one file, and 3d variables for atmosphere either in one file or one variable per file, as 3d variables can be very large.

File: era5_monthly_averages_single_level_1979_2024/README.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+1. `mean_surface_downward_long_wave_radiation_flux`
+2. `mean_surface_downward_short_wave_radiation_flux`
+3. `mean_surface_latent_heat_flux`
+4. `mean_surface_net_long_wave_radiation_flux`
+5. `mean_surface_net_short_wave_radiation_flux`
+6. `mean_surface_sensible_heat_flux`
+7. `mean_surface_runoff_rate`
+8. `mean_sub_surface_runoff_rate`
+9. `total_column_water`
+10. `number`
```
Comment (kmdeck) [inline era5_monthly_averages_single_level_1979_2024/README.md:39]: we can define this too?

File: era5_monthly_averages_single_level_1979_2024/process_downloads.jl [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+
+    defVar(
+        output_ds,
+        "longitude",
+        input_ds["longitude"][:],
+        ("longitude",);
+        attrib = input_ds["longitude"].attrib,
+    )
+
+    new_times = map(input_ds["date"][:]) do t
```
Comment (kmdeck) [inline era5_monthly_averages_single_level_1979_2024/process_downloads.jl:116]: Could you explain what lines 116-125 do?

File: era5_monthly_averages_single_level_1979_2024/process_downloads.jl [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+        @assert new_times[i] != new_times[i-1]
+    end
+    new_times_attribs = ["standard_name" => "time", "long_name" => "Time"]
+
+    defVar(output_ds, "time", new_times, ("time",); attrib = new_times_attribs)
+    close(output_ds)
+end
+
+convert_to_f32(x) = Float32(x)
+convert_to_f32(x::Missing) = x
```
Comment (kmdeck) [inline era5_monthly_averages_single_level_1979_2024/process_downloads.jl:133]: Out of curiosity, when was there missing data?

Comment (kmdeck) [reply era5_monthly_averages_single_level_1979_2024/process_downloads.jl:133]: maybe runoff over ocean?

**Review by kmdeck — APPROVED** [review]
I think this looks good! Thank you! We can defer to @szy21 regarding the files

**Comment by imreddyTeja** [comment]
I just noticed the `times` in the monthly averages (non hourly) are formatted wrong. I'll fix them and split up the files

File: era5_monthly_averages_single_level_1979_2024/README.md [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+This variable is only defined on the time dimension, and a value of 5 indicates any data at that point on the time axis is from ERA5T, which is an initial release of ERA5 data. A value of 1 indicates that data for that time is no longer potentially subject to change. No preprocessing is done other than removing residual GRIB attributes. It is not included in the hourly dataset.
+
+## Temporal Coverage
+
+The two output files have different temporal coverage.
+
+### `era5_monthly_surface_fluxes_197901-202410.nc`
+
+This file contains Monthly averaged reanalysis from 1979 to present (October 2024 at time of creation), which is produced by averaging all daily data for each month. This results in 12*(2024-1979 + 10/12) = 550 points on the
+time dimension, where each point is the 15th of the month that the point represents. For example, the 6th index of `time` is `19790601-01-15T00:00:00`,
```
Comment (szy21) [inline era5_monthly_averages_single_level_1979_2024/README.md:103]:
```suggestion
time dimension, where each point is the 15th of the month that the point represents. For example, the 6th index of `time` is `1979-06-15T00:00:00`,
```

File: era5_monthly_averages_single_level_1979_2024/process_downloads.jl [hunk 5]
```diff
@@ -0,0 +1,319 @@
...
+    defDim(output_ds, "latitude", input_ds.dim["latitude"])
+    ignore_mod_31 = [1, 3, 5, 7, 9, 11, 13]
+    time_indx = filter(i -> !(i % 31 in ignore_mod_31), 1:length(input_ds["time"][:]))
+    defDim(output_ds, "time", length(time_indx))
+    missing_indices = filter(i -> (i % 31 in ignore_mod_31), 1:length(input_ds["time"][:]));
+    for index in missing_indices
+        if !all(input_ds["tcw"][:,:,index] .=== missing)
+            @error "The index pattern of the invalid data is not as expected"
+        end
+    end
```
Comment (szy21) [inline era5_monthly_averages_single_level_1979_2024/process_downloads.jl:158]: What does this block of code do?

**Review by szy21 — APPROVED** [review]
Thanks!

File: era5_monthly_averages_single_level_1979_2024/process_downloads.jl (hunk 3 above)
Comment (imreddyTeja) [reply era5_monthly_averages_single_level_1979_2024/process_downloads.jl]: Downloading Monthly averages by hour of day results in some weirdness with the time dimension, where there are duplicate entries. Some variables are defined on one of the duplicates and some on the other. I had improperly dealt with this (assumed all were on one of the duplictes), which resulted in missing data. This is fixed now, and there is no missing data

File: era5_monthly_averages_single_level_1979_2024/process_downloads.jl (hunk 5 above)
Comment (imreddyTeja) [reply era5_monthly_averages_single_level_1979_2024/process_downloads.jl]: Downloading Monthly averages by hour of day results in some weirdness with the time dimension, where there are duplicate entries. Some variables are defined on one of the duplicates and some on the other. This removed the duplicate points which only hold missing data. I added a comment describing this.

**Pull request merged**